src/PhpParser/Comparing/ConditionSearcher.php

```diff
@@ -67,7 +75,11 @@ private function searchElseForVariableRedeclaration(Assign $assign, Else_ $else)
                 continue;
             }
 
-            return true;
+            $isFoundInExpr = (bool) $this->betterNodeFinder->findFirst(
+                $statementElse->expr->expr,
+                fn (Node $node): bool => $this->nodeComparator->areNodesEqual($varNode, $node)
+            );
+            return ! $isFoundInExpr;
```

Comment:
Why not `if ($isFoundInExpr) { continue; } else { return true }` ?

Reply:
implemented https://github.com/rectorphp/rector-src/pull/205/commits/e598791ca258546ac6ea0f5050976f71e728581d

Reply:
I thought about it and I think it was better like this, since we only really care about the first statement that makes it to here.